src/lager.erl

```diff
@@ -126,6 +137,11 @@ do_log(Severity, Metadata, Format, Args, Size, SeverityAsInt, LevelThreshold, Tr
 dispatch_log(Severity, _Module, _Function, _Line, _Pid, Metadata, Format, Args, Size) ->
     dispatch_log(Severity, Metadata, Format, Args, Size).
 
+
+%% TODO:
```

Comment:
I hate the `log2` name, but I couldn't think of a better way to handle this. It feels like there **ought** to be a function with arity 4 that takes Sink, Level, Pid and Message. Open to naming suggestions here.